src/oss/confluent.sh

```diff
@@ -1433,6 +1489,52 @@ version_command() {
     fi
 }
 
+produce_usage() {
+    cat <<EOF
+Usage: ${command_name} <topicname> [--value-format avro] [other optional args]
+
+Description:
+    Produce to Kafka topic specified by <topicname>.
+
+    Without '--value-format avro', it calls the 'kafka-console-producer' command,
+    automatically with '--topic <topicname> --broker-list localhost:9092' and passes in <other optional args>.
+
+    With '--value-format avro', it calls the 'kafka-avro-console-producer' command,
+    automatically with '--topic <topicname> --broker-list localhost:9092' and passes in <other optional args>.
+    This requires an additional argument '--property value.schema=<schema>'
+
+Examples:
+    confluent produce mytopic1
+
+    confluent produce mytopic1 --value-format avro --property value.schema='{"type":"record","name":"myrecord","fields":[{"name":"f1","type":"string"}]}'
+
+EOF
+    exit 0
```

Comment:
Usage should exit with 1 when called from a code that found a missing/invalid parameter, such as with the topic name. So maybe pass the exit code as a parameter to produce_usage()

Reply:
@edenhill , this code follows convention of the rest of Confluent CLI.  It would have to be changed everywhere

Reply:
Well, I insist :)
A failing command must have a non-zero exit status (typically 1), otherwise it can't be used programmatically in other scripts.

Reply:
@edenhill : turns out not such a big change.  I have fixed this in the latest commit

Reply:
We have `die` for abrupt interruption with a message and a saner exit code.

Reply:
The usage is printed twice, once intentionally with `help` and once on error if users forgets `topicname`.  So we can't always die in the usage.  The current code exits with the appropriate exit code.